tests/semanticdb/metac.expect

```diff
@@ -4217,7 +4271,7 @@ flags/p/package.y(). => protected implicit var method y Int
 flags/p/package.z(). => method z (param pp: Int): Int
 flags/p/package.z().(pp) => param pp: Int
 local0 => val local xs2: Nothing
-local1 => case val method t 
+local1 => case type t 
```

Comment:
would this be able to drop `case` or does scala 2 do the same? e.g. when you have a wildcard type in a normal pattern match

Reply:
In scala2, `t` of `case m: List[t]` gets `abstract type t`. I'm trying to remove `case` from them :)

<details>
<summary>full diff</summary>

```diff
diff --git a/semanticdb/integration/src/main/scala/example/Advanced.scala b/semanticdb/integration/src/main/scala/example/Advanced.scala
index 32fc9a731..1af078cb9 100644
--- a/semanticdb/integration/src/main/scala/example/Advanced.scala
+++ b/semanticdb/integration/src/main/scala/example/Advanced.scala
@@ -44,4 +44,5 @@ object Test {
         ()
     }
   }
+  def foo(m: Any) = m match { case m: List[t] => ??? }
 }
diff --git a/tests/jvm/src/test/resources/example/Advanced.scala b/tests/jvm/src/test/resources/example/Advanced.scala
index 7b5a319fa..8399799bc 100644
--- a/tests/jvm/src/test/resources/example/Advanced.scala
+++ b/tests/jvm/src/test/resources/example/Advanced.scala
@@ -44,4 +44,5 @@ object Test/*<=advanced.Test.*/ {
         ()
     }
   }
+  def foo/*<=advanced.Test.foo().*/(m/*<=advanced.Test.foo().(m)*/: Any/*=>scala.Any#*/) = m/*=>advanced.Test.foo().(m)*/ match { case m/*<=local14*/: List/*=>scala.package.List#*/[t/*<=local15*/] => ???/*=>scala.Predef.`???`().*/ }
 }
diff --git a/tests/jvm/src/test/resources/metac.expect b/tests/jvm/src/test/resources/metac.expect
index 0bebf76c6..05a6d7185 100644
--- a/tests/jvm/src/test/resources/metac.expect
+++ b/tests/jvm/src/test/resources/metac.expect
@@ -59,8 +59,8 @@ Schema => SemanticDB v4
 Uri => semanticdb/integration/src/main/scala/example/Advanced.scala
 Text => non-empty
 Language => Scala
-Symbols => 44 entries
-Occurrences => 110 entries
+Symbols => 48 entries
+Occurrences => 118 entries
 Synthetics => 1 entries
 
 Symbols:
@@ -110,7 +110,7 @@ advanced/Structural#s3(). => method s3: AnyRef { def m(x: Int): Int }
   m => local3
   x => local4
   Int => scala/Int#
-advanced/Test. => final object Test extends AnyRef { +13 decls }
+advanced/Test. => final object Test extends AnyRef { +14 decls }
   AnyRef => scala/AnyRef#
 advanced/Test.e. => val method e: Existential
   Existential => advanced/Existential#
@@ -128,6 +128,12 @@ advanced/Test.e2x. => val method e2x: List[Any]
   Any => scala/Any#
 advanced/Test.e2xx. => val method e2xx: Any
   Any => scala/Any#
+advanced/Test.foo(). => method foo(m: Any): Nothing
+  m => advanced/Test.foo().(m)
+  Any => scala/Any#
+  Nothing => scala/Nothing#
+advanced/Test.foo().(m) => param m: Any
+  Any => scala/Any#
 advanced/Test.s. => val method s: Structural
   Structural => advanced/Structural#
 advanced/Test.s1. => val method s1: AnyRef { val def x: Int }
@@ -174,6 +180,10 @@ local11 => val local e3: List[_]
 local12 => abstract type _
 local13 => val local e3x: _
   _ => local12
+local14 => val local m: List[t]
+  List => scala/package.List#
+  t => local15
+local15 => abstract type t
 
 Occurrences:
 [0:8..0:16): advanced <= advanced/
@@ -286,6 +296,14 @@ Occurrences:
 [42:12..42:15): e3x <= local13
 [42:18..42:20): e3 => local11
 [42:21..42:25): head => scala/collection/IterableLike#head().
+[46:6..46:9): foo <= advanced/Test.foo().
+[46:10..46:11): m <= advanced/Test.foo().(m)
+[46:13..46:16): Any => scala/Any#
+[46:20..46:21): m => advanced/Test.foo().(m)
+[46:35..46:36): m <= local14
+[46:38..46:42): List => scala/package.List#
+[46:43..46:44): t <= local15
+[46:49..46:52): ??? => scala/Predef.`???`().
 
 Synthetics:
 [39:2..39:9): locally => *[Unit]
diff --git a/tests/jvm/src/test/resources/metacp.expect b/tests/jvm/src/test/resources/metacp.expect
index 912135494..6f0ff6645 100644
--- a/tests/jvm/src/test/resources/metacp.expect
+++ b/tests/jvm/src/test/resources/metacp.expect
@@ -118,10 +118,10 @@ Schema => SemanticDB v4
 Uri => advanced/Test.class
 Text => empty
 Language => Scala
-Symbols => 14 entries
+Symbols => 16 entries
 
 Symbols:
-advanced/Test. => final object Test extends AnyRef { +13 decls }
+advanced/Test. => final object Test extends AnyRef { +14 decls }
   AnyRef => scala/AnyRef#
 advanced/Test.e. => val method e: Existential
   Existential => advanced/Existential#
@@ -139,6 +139,12 @@ advanced/Test.e2x. => val method e2x: List[Any]
   Any => scala/Any#
 advanced/Test.e2xx. => val method e2xx: Any
   Any => scala/Any#
+advanced/Test.foo(). => method foo(m: Any): Nothing
+  m => advanced/Test.foo().(m)
+  Any => scala/Any#
+  Nothing => scala/Nothing#
+advanced/Test.foo().(m) => param m: Any
+  Any => scala/Any#
 advanced/Test.s. => val method s: Structural
   Structural => advanced/Structural#
 advanced/Test.s1. => val method s1: AnyRef { val def x: Int }
```

</details>

Reply:
They become `case` because the symbols of `t` has `CaseClass` flag, and props added here ↓
https://github.com/lampepfl/dotty/blob/28d59a07b0550e4da94e0e835d7efc6961ce5171/compiler/src/dotty/tools/dotc/semanticdb/Scala3.scala#L312-L313

Maybe we should remove `CaseClass` flag from the symbol 🤔